source/common/stats/tag_extractor_impl.h

```diff
@@ -14,6 +18,39 @@
 namespace Envoy {
 namespace Stats {
 
+// To check if a tag extractor is actually used you can run
+// bazel test //test/... --test_output=streamed --define=perf_annotation=enabled
+#ifdef ENVOY_PERF_ANNOTATION
+
+struct Counters {
+  uint32_t skipped_{};
+  uint32_t matched_{};
+  uint32_t missed_{};
+};
+
+#define PERF_TAG_COUNTERS(var)                                                                     \
+  ~TagExtractorImplBase() override {                                                               \
+    std::cout << fmt::format("TagStats for {} tag extractor: skipped {}, matched {}, missing {}",  \
+                             name_, var->skipped_, var->matched_, var->missed_)                    \
+              << std::endl;                                                                        \
+  }                                                                                                \
+  std::unique_ptr<Counters> var
+
+#define PERF_TAG_COUNTERS_INIT(var) var = std::make_unique<Counters>()
```

Comment:
Nit: wrap the expansion in parens or do/while for safety.

Reply:
Out of curiosity, what's the scenario where this would parse unexpectedly?

Usually I'd use do/while in a macro if there were multiple statements in the expansion.

Reply:
What if I make the macros non-parameterized as they are used only once?

Reply:
Yes that makes sense. You could also have
```
#define PERF_TAG_INC(member) ++counters->member
```
and then you only need one macro rather than one each for skipped/missed/matched, but I am not too concerned about this; whatever you prefer.

Reply:
Thanks! This simplifies the code a bit. Updated.